services/frontend-service/src/ui/components/ProductVersion/ProductVersion.tsx

```diff
@@ -93,11 +144,18 @@ export const ProductVersion: React.FC = () => {
         if (tagsResponse.response.tagData.length > 0) {
             setShowSummarySpinner(true);
             const env = splitCombinedGroupName(environment);
-            getSummary(tagsResponse.response.tagData[0].commitId, env[0], env[1]);
+            if (selectedTag === '') {
+                setSelectedTag(tagsResponse.response.tagData[0].commitId);
+                searchParams.set('tag', tagsResponse.response.tagData[0].commitId);
+                setSearchParams(searchParams);
+                getSummary(tagsResponse.response.tagData[0].commitId, env[0], env[1]);
+            } else {
+                getSummary(selectedTag, env[0], env[1]);
+            }
             setDisplayVersion(true);
-            setSelectedTag(tagsResponse.response.tagData[0].commitId);
         }
-    }, [tagsResponse, envGroupResponse, searchParams, environment]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Not seeing the need to disable this rule here. Technically we have less dependencies than before, no?

Reply:
There's some dependencies that disrupt with testing that I was unable to work around and be able to test properly. Without disabling the rule, I can't test the page and it is forever stuck in a loading state.

Reply:
@gsandok assuming we do not find a way to set the deps properly: Your comment here (in GitHub) should be part of the source code comment, otherwise it will get lost.

I've never seen a case, where the deps could not be set correctly, let's talk directly @gsandok .